python/ray/train/context.py

```diff
@@ -0,0 +1,95 @@
+import threading
+from typing import TYPE_CHECKING, Optional
+
+from ray.air import Checkpoint
+from ray.train._internal import session
+from ray.util.annotations import PublicAPI
+
+
+if TYPE_CHECKING:
+    from ray.data import DataIterator
+    from ray.tune.execution.placement_groups import PlacementGroupFactory
+
+
+# The context singleton on this process.
+_default_context: "Optional[Context]" = None
+_context_lock = threading.Lock()
+
+
+def _copy_doc(copy_func):
+
+    def wrapped(func):
+        func.__doc__ = copy_func.__doc__
+        return func
+
+    return wrapped
+
+
+@PublicAPI(stability="beta")
+class Context:
```

Comment:
Should we explicitly name this `TrainContext` (cf `RuntimeContext`, `WorkerContext`)?

Reply:
Yes, I renamed it now :)